src/index.js

```diff
@@ -0,0 +1,13 @@
+import { displayCategoryLink, searchCategory, searchLinks } from "./displayUI";
+import { instanceOfMainClass } from './mainClass';
+import { transform } from './transform';
+
+instanceOfMainClass.allCategories = searchCategory;
+instanceOfMainClass.searchThroughLinks = searchLinks;
+instanceOfMainClass.linksUpdate = displayCategoryLink;
```

Comment:
Still not added encapsulation to set callbacks, you directly set properties, not bad. But I would like to learn litle bit about OOP encapsulation and about private access/setters/getters